app/views/contexts/summary/question.py

```diff
@@ -18,22 +27,43 @@ def __init__(
             question_schema.get("title") or question_schema["answers"][0]["label"]
         )
         self.number = question_schema.get("number", None)
+
         self.rule_evaluator = rule_evaluator
 
-        self.answers = self._build_answers(answer_store, question_schema)
+        self.answers = self._build_answers(
+            answer_store=answer_store,
+            question_schema=question_schema,
+            block_id=block_id,
+            list_name=location.list_name if location else None,
+            return_to=return_to,
+        )
 
     def _get_answer(self, answer_store, answer_id):
         return answer_store.get_escaped_answer_value(answer_id, self.list_item_id)
 
-    def _build_answers(self, answer_store, question_schema):
+    def _build_answers(
+        self, *, answer_store, question_schema, block_id, list_name, return_to
+    ):
 
         if self.summary:
+            answer_id = f"{self.id}-concatenated-answer"
```

Comment:
I think this needs a functional test to ensure anchor uses this. As it is explicitly different to other answer types. It does work functionally, but I don't think we have any tests to validate this?